File: stack/min_stack.py
class MinStack:
    def __init__(self):
        self.min_stack = []
        self.stack = []

    def push(self, x: int) -> None:
        current_min = self.min_stack[-1] if self.min_stack else float('inf')
        self.stack.append(x)
        self.min_stack.append(min(current_min, x))

    def pop(self) -> None:
        self.min_stack.pop()
        self.stack.pop()

    def top(self) -> int:
        return self.stack[-1]

    def getMin(self) -> int:
        return self.min_stack[-1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: stack/min_stack.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class MinStack
{
    private Stack<int> stack;
    private Stack<int> minStack;

    /** initialize your data structure here. */
    public MinStack()
    {
        stack = new Stack<int>();
        minStack = new Stack<int>();
    }

    public void Push(int x)
    {
        int currentMin = minStack.Any() ? minStack.Peek() : int.MaxValue;

        stack.Push(x);
        minStack.Push(Math.Min(currentMin, x));
    }

    public void Pop()
    {
        minStack.Pop();
        stack.Pop();
    }

    public int Top()
    {
        return stack.Peek();
    }

    public int GetMin()
    {
        return minStack.Peek();
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.Push(x);
 * obj.Pop();
 * int param_3 = obj.Top();
 * int param_4 = obj.GetMin();
 */
